File: Day214/repeatedSubstringPattern.go
package main

import (
	"fmt"
	"strings"
)


func rsp(str string) bool {
	n := len(str)
	for i:=n/2; i>0; i-- {
		if n%i==0 {
			temp := str[:i]
			var sb string
			len := n / i
			for j:=0; j<len; j++ {
				sb += temp
			}
			if sb == str {
				return true
			}
		}
	}
	return false
}

func orsp(str string) bool {
	n := len(str)
	for i:= n/2; i>0; i-- {
		if n%i == 0 && str[i-1] == str[n - 1] {
			temp := str[:i]
			len := n /i
			if strings.Repeat(temp, len) == str {
				return true
			}
		}
	}
	return false
}

func main() {
	str := "abcabcabcabc"
	fmt.Println(rsp(str))
}

File: Day214/repeatedSubstringPattern.java
package Day214;
class repeatedSubstringPattern {
    public static boolean rsp(String s) {
        int n = s.length();
        for (int i=n/2; i>0; i--) {
            if (n % i == 0) {
                String temp = s.substring(0, i);
                StringBuilder sb = new StringBuilder();
                int len = n / i;
                for (int j=0; j<len; j++) {
                    sb.append(temp);
                }
                if (sb.toString().equals(s)) {
                    return true;
                }
            }
        }
        return false;
  
    }
    public static boolean orsp(String str) {
        int len = str.length();
        for (int i=len/2; i>0; i--) {
            if (len%2==0 && str.charAt(i-1) == str.charAt(len-1)) {
                String temp = str.substring(0, i);
                int l = len / i;
                if (temp.repeat(l).equals(str)) {
                    return true;
                }
            }
        }
        return false;
    }
    public static void main(String[] args) {
        String str = "abcabcabcabc";
        System.out.println(rsp(str));
    }
}